apps/portal/prisma/schema.prisma

```diff
@@ -162,4 +171,136 @@ model ResumesCommentVote {
+    // role       String
+    createdAt  DateTime @default(now())
+    updatedAt  DateTime @updatedAt
+
+    user     User?             @relation(fields: [userId], references: [id], onDelete: SetNull)
+    question QuestionsQuestion @relation(fields: [questionId], references: [id], onDelete: Cascade)
+}
+
+model QuestionsQuestionVote {
+    id         String        @id @default(cuid())
+    questionId String
+    userId     String?
+    vote       QuestionsVote
+    createdAt  DateTime      @default(now())
+    updatedAt  DateTime      @updatedAt
+
+    user     User?             @relation(fields: [userId], references: [id], onDelete: SetNull)
+    question QuestionsQuestion @relation(fields: [questionId], references: [id], onDelete: Cascade)
+
+    @@unique([questionId, userId])
+}
+
+model QuestionsQuestionComment {
+    id         String   @id @default(cuid())
+    questionId String
+    userId     String?
+    content    String   @db.Text
+    createdAt  DateTime @default(now())
+    updatedAt  DateTime @updatedAt
+
+    user                         User?                          @relation(fields: [userId], references: [id], onDelete: SetNull)
+    question                     QuestionsQuestion              @relation(fields: [questionId], references: [id], onDelete: Cascade)
+    QuestionsQuestionCommentVote QuestionsQuestionCommentVote[]
+}
+
+model QuestionsQuestionCommentVote {
+    id                String        @id @default(cuid())
+    questionCommentId String
+    userId            String?
+    vote              QuestionsVote
+    createdAt         DateTime      @default(now())
+    updatedAt         DateTime      @updatedAt
+
+    user            User?                    @relation(fields: [userId], references: [id], onDelete: SetNull)
+    questionComment QuestionsQuestionComment @relation(fields: [questionCommentId], references: [id], onDelete: Cascade)
+
+    @@unique([questionCommentId, userId])
+}
+
+model QuestionsAnswer {
+    id         String   @id @default(cuid())
+    questionId String
+    userId     String?
+    content    String   @db.Text
+    createdAt  DateTime @default(now())
+    updatedAt  DateTime @updatedAt
+
+    user                   User?                    @relation(fields: [userId], references: [id], onDelete: SetNull)
+    question               QuestionsQuestion        @relation(fields: [questionId], references: [id], onDelete: Cascade)
+    QuestionsAnswerVote    QuestionsAnswerVote[]
+    QuestionsAnswerComment QuestionsAnswerComment[]
+}
+
+model QuestionsAnswerVote {
+    id        String        @id @default(cuid())
+    answerId  String
+    userId    String?
+    vote      QuestionsVote
+    createdAt DateTime      @default(now())
+    updatedAt DateTime      @updatedAt
+
+    user   User?           @relation(fields: [userId], references: [id], onDelete: SetNull)
+    answer QuestionsAnswer @relation(fields: [answerId], references: [id], onDelete: Cascade)
+
+    @@unique([answerId, userId])
```

Comment:
Let's allow a user to submit more than one answer. It'd be useful for some questions where there are multiple answers

Reply:
I believe this is the upvote struct, the intention is to ensure there's a unique vote for each user for each question. Does that make sense?

Reply:
My bad, you're right.